scanner/Makefile

```diff
@@ -88,6 +78,26 @@ $(build-t): deps
 	@echo "+ $@"
 	$(build-cmd) -o $@ ./cmd/$(@F)
 
+.PHONY: build-local
+build-local:
+	@echo "+ $@"
+	$(MAKE) SILENT=$(SILENT) GOOS=${HOST_OS} build
+
+.PHONY: build-local-scanner
+build-local-scanner:
+	@echo "+ $@"
+	$(MAKE) SILENT=$(SILENT) GOOS=${HOST_OS} bin/scanner
+
+.PHONY: build-local-scannerctl
+build-local-scannerctl:
+	@echo "+ $@"
+	$(MAKE) SILENT=$(SILENT) GOOS=${HOST_OS} bin/scannerctl
+
+.PHONY: build-local-updater
+build-local-updater:
+	@echo "+ $@"
+	$(MAKE) SILENT=$(SILENT) GOOS=${HOST_OS} bin/updater
+
```

Comment:
Let's do this:

1. Let's default `GOOS` to `$(HOST_OS)`. I.e., don't use `DEFAULT_GOOS` anymore by default.
2. When building images, let's set `GOOS=$(DEFAULT_GOOS)` when building `bin/scanner`.

With the above, everything is `build-local`, except when building images, which is when we need `bin/scanner` to be built with `GOOS=linux` ("linux" is `DEFAULT_GOOS`).

Here:

```diff
diff --git a/scanner/Makefile b/scanner/Makefile
index e05c17c280..ad06331cdc 100644
--- a/scanner/Makefile
+++ b/scanner/Makefile
@@ -25,7 +25,6 @@ endif
 UNAME_S := $(shell uname -s)
 UNAME_M := $(shell uname -m)
 
-GOOS := $(DEFAULT_GOOS)
 HOST_OS := linux
 ifeq ($(UNAME_S),Darwin)
 	HOST_OS := darwin
@@ -34,11 +33,12 @@ endif
 GO_VERSION := $(firstword $(subst go version ,,$(shell go version)))
 EXPECTED_GO_VERSION := $(shell cat ../EXPECTED_GO_VERSION)
 
+GOOS := $(HOST_OS)
 
-GO_BUILD_FLAGS := CGO_ENABLED=0 GOOS=$(GOOS) GOARCH=$(GOARCH)
-GO_BUILD_CMD := $(GO_BUILD_FLAGS) go build \
-                 -trimpath \
-                 -ldflags="-X github.com/stackrox/stackrox/scanner/internal/version.Version=$(TAG)"
+GO_BUILD_FLAGS = CGO_ENABLED=0 GOOS=${GOOS} GOARCH=${GOARCH}
+GO_BUILD_CMD   = $(GO_BUILD_FLAGS) go build \
+                   -trimpath \
+                   -ldflags="-X github.com/stackrox/stackrox/scanner/internal/version.Version=$(TAG)"
 GO_TEST_CMD := $(GO_BUILD_FLAGS) go test
 
 DOCKERBUILD := $(CURDIR)/../scripts/docker-build.sh
@@ -76,7 +76,7 @@ tag:
 ## Go Builds ##
 ###############
 
-build-cmd := $(GO_BUILD_CMD)
+build-cmd = $(GO_BUILD_CMD)
 build-d   := bin
 build-t   := $(addprefix $(build-d)/,$(notdir $(wildcard cmd/*)))
 
@@ -102,6 +102,7 @@ image-prefix := scanner-v4
 .PHONY: images
 images: scanner-image db-image
 
+image/scanner/bin/scanner: GOOS=$(DEFAULT_GOOS)
 image/scanner/bin/scanner: bin/scanner
 	@echo "+ $@"
 	cp $< $@
```

Reply:
I did this suggestion plus I updated `GO_TEST_CMD := $(GO_BUILD_FLAGS) go test` to `GO_TEST_CMD = $(GO_BUILD_FLAGS) go test`

Reply:
LGTM.